### ISSUE TITLE
Nuget restore in UwpAppxBuildTools.BuildAppxAsync doesn't do anything, causing build to fail

### ISSUE THREAD

**Issue by ChristopherManthei** [microsoft/MixedRealityToolkit-Unity#8607]
## Describe the bug

If a project has nuget references, using MRTK to build it will fail as NuGet packages are not restored before building.
The problem is in **UwpAppxBuildTools.BuildAppxAsync** which calls 
```MSBuild.exe" "[PROJECT_SLN_PATH]" /t:restore```
However, this will output something similiar to this:
```
Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
Build started 21/09/2020 16:05:46.
Project "E:\ShowcaseBuilder\ARR Showcase.sln" on node 1 (Restore target(s)).
ValidateSolutionConfiguration:
  Building solution configuration "Debug|ARM".
_GetAllRestoreProjectPathItems:
  Determining projects to restore...
Restore:
  Nothing to do. None of the projects specified contain packages to restore.
Done Building Project "E:\ShowcaseBuilder\ARR Showcase.sln" (Restore target(s)).
```
And then it continues to building which fails with
```
E:\ShowcaseBuilder\ARR Showcase\ARR Showcase.vcxproj(270,5): error : This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\packages\Microsoft.Azure.SpatialAnchors.WinCPP.Redist.2.2.0\build\native\Microsoft.Azure.SpatialAnchors.WinCPP.Redist.targets.
```
Instead of using MSBuild.exe, nuget.exe should be used. E.g. ```nuget.exe restore  "[PROJECT_SLN_PATH]"```.

## To reproduce

1. Create a new project
2. Add e.g. **Azure Spatial Anchors** package to it.
3. Add some custom script to call UwpAppxBuildTools.BuildAppxAsync and run it
4. See build error in log

While a bit big, you can use the [ARR Showcase](https://github.com/Azure/azure-remote-rendering/tree/master/Unity/AzureRemoteRenderingShowcase/arr-showcase-app) app as an example:
1. Checkout the above repo
2. Open **Unity/AzureRemoteRenderingShowcase/arr-showcase-app** project
3. Click **Menu -> Builder -> Build HoloLens 2 Client (arm)**
4. See build error in log

## Expected behavior

NuGet packages are restored and build succeeds.

## Your setup (please complete the following information)

- Unity Version [2019.3.1f1]
- MRTK Version [v2.2.0]

## Target platform (please complete the following information)
- HoloLens 2

**Comment by keveleigh** [comment]
@MaxWang-MS Is this the issue that can be solved by your PR https://github.com/microsoft/MixedRealityToolkit-Unity/pull/9556?

**Comment by MaxWang-MS** [comment]
Fixed in #9556.
